src/Npgsql/Replication/PgOutput/Messages/RollbackPreparedMessage.cs

```diff
@@ -0,0 +1,65 @@
+﻿using NpgsqlTypes;
+using System;
+
+namespace Npgsql.Replication.PgOutput.Messages;
+
+/// <summary>
+/// Logical Replication Protocol rollback prepared message
+/// </summary>
+public sealed class RollbackPreparedMessage : PreparedTransactionMessageBase
+{
+    /// <summary>
+    /// Flags for the rollback prepared; currently unused.
+    /// </summary>
+    public RollbackPreparedFlags Flags { get; private set; }
+
+    /// <summary>
+    /// The LSN of the rollback prepared.
```

Comment:
Is this the field [documented](https://www.postgresql.org/docs/devel/protocol-logicalrep-message-formats.html) as "The end LSN of the rollback prepared transaction" (note "end")? It's a bit confusing as there are two "end" LSNs in this message - we may want to change the property names and comments.

It would also be somewhat weird to expose the internal `StartLsn` as an "end" property... We may want to rename the internal properties "FirstLsn" and "SecondLsn" or similar.

Reply:
> Is this the field [documented](https://www.postgresql.org/docs/devel/protocol-logicalrep-message-formats.html) as "The end LSN of the rollback prepared transaction" (note "end")?

Almost. It is documented as "The end LSN of the prepared transaction." but yes, that's a mistake (probably copy-paste error) that needs fixing and I agree with all you said above.